Merge from wnprc18.3 svn r.64261 to r.65647

labkey-bpatel opened the pull request:
#### Rationale
Merge from wnprc18.3 svn r.64261 to r.65647 (skipped r. 64864 & 65166 since these were backports)

#### Related Pull Requests
https://github.com/LabKey/commonAssays/pull/200
LabKey/ehrModules#56
LabKey/onprcEHRModules#20
LabKey/platform#1370
LabKey/tnprc_billing#72

labkey-bpatel committed:
Merge from wnprc18.3 svn r.64261-64312

labkey-bpatel committed:
Merge from wnprc18.3 svn r.64313-64598

bbimber commented on LDK/resources/web/LDK/panel/TabbedReportPanel.js:
```diff
@@ -764,19 +764,34 @@ Ext4.define('LDK.panel.TabbedReportPanel', {
-        }
-
-        //allow any other supported properties to be applied through here
-        if (tab.report.queryConfig){
```
The idea of iterating the whitelist of configOptions is a good one, but are you sure no code uses report.queryConfig? It doesnt appear that's backward compatible

labkey-bpatel replied:
Good point, i just did a text search in LabKey repo, and did not see usages for report.queryConfig. Also, will keep an eye on any test failures around this, if any.

bbimber commented on LDK/resources/web/LDK/Utils.js:
```diff
@@ -556,7 +560,9 @@ LDK.Utils = new function(){
 
             if (subjectArray.length > 0) {
-                subjectArray = Ext4.unique(subjectArray);
+                if (removeDuplicates) {
```
admittedly picky, but cant you put if if(!!removeDuplicates)? and skip the removeDuplicates==null check above?

labkey-jeckels replied:
For the case in question, that would be !!null, which evaluates to false. We want null to be transformed to true.

bbimber replied:
yes, clearly a typo on my part.  anyway, I think the reason this struck me was party that it's using javascript's sloppy comparisons.  under the condition it's really testing (no argument passed), isnt 'undefined' the value we expect to get?  we have this awkward/imprecise comparison (null is not really undefined).

I assume there are probably some callers elsewhere, but wouldnt making the argument 'preserveDuplicates' be more elegant?  This way the no-arg / default case would be false, which is our current behavior.  We just need to test "if(!preserveDuplicates)"

labkey-jeckels replied:
I flipped the condition

bbimber commented on LDK/test/src/org/labkey/test/util/external/labModules/LabModuleHelper.java:
```diff
@@ -50,9 +51,21 @@
     public LabModuleHelper(BaseWebDriverTest test)
     {
         _test = test;
+
```
I'm confused - why is this in the constructor?

labkey-bpatel replied:
Confusing indeed - this was merge gone bad, now remove. Good catch!

bbimber commented on LDK/test/src/org/labkey/test/util/external/labModules/LabModuleHelper.java:
```diff
@@ -426,4 +439,50 @@ public void initiatePipelineJob(String importAction, List<String> files, String
         _test._fileBrowserHelper.selectImportDataAction(importAction);
     }
+
+    public void defineViralAssayWithAdditionalFields(String provider, String label, Map<String, String> batchFields, Map<String, String> runFields, Map<String, String> resultFields)
```
Why is this in the helper class, and not in the actual test (like ViralLoadAssayTest or whatever uses this)

labkey-bpatel replied:
Not too sure, @labkey-jeckels any idea?
Looks like this was part of 17.2 merges to wnprc18.3 branch.

labkey-jeckels replied:
Now moved.

bbimber commented on laboratory/resources/web/laboratory/panel/AbstractAssayPanel.js:
```diff
@@ -197,6 +197,17 @@ Ext4.define('Laboratory.panel.AbstractAssayPanel', {
             }
         }
 
+        if (fieldObj.getInitialValue){
```
Doesnt this addition just duplicate the code right above it?

labkey-bpatel replied:
Yup, now removed - looks like it happened as part of 17.2 merges to wnprc18.3 branch.

bbimber reviewed (commented):
I realize I'm jumping in here quickly after github sent the notification and there may or may not have been more review on this.  most seems fine, but there are several proposed changes in here that really should have gotten triaged.

labkey-bpatel committed:
Fix merge.

labkey-bpatel committed:
Remove duplicate code.

labkey-jeckels committed:
Various test fixes, including:

Customize source_material query directly instead of trying to mutate a locked FK lookup table

Update to use new assay designer UI

Migrate <javaCustomizer> syntax

labkey-jeckels committed:
Use ExtJS4 instead of 3

bbimber commented on LDK/test/src/org/labkey/test/util/external/labModules/LabModuleHelper.java:
```diff
@@ -426,4 +434,34 @@ public void initiatePipelineJob(String importAction, List<String> files, String
 
         _test._fileBrowserHelper.selectImportDataAction(importAction);
     }
+
```
Same comment as originally posted: does this viral load assay code belong in LabModuleHelper, or can it live in ViralLoadAssayTest?  Where's it get used?  I dont see reference to 'viral' in LabModulesTest or LabModuleHelper in 20.7.

labkey-jeckels replied:
I just moved the now-working code into ViralLoadAssayTest

bbimber replied:
thanks

labkey-jeckels committed:
Cascade user delete though ehr_billing.dataaccess

Move code from LabModuleHelper to only place that uses it, ViralLoadAssayTest

Fix rate import codepath in WNPRC_EHRTest

labkey-jeckels committed:
Remove other parts of the merge

labkey-jeckels committed:
Flip meaning of argument at controls de-duplication of IDs

Fix up handling of drop-downs in WNPRC misc charges entry

bbimber committed:
Update AssayTemplatePanel.js

Consistent line spacing for parentheses

bbimber commented:
@labkey-jeckels I made one minor commit to AssayTemplatePanel to make line spacing consistent for parentheses. I also think I closed all the conversations I need as part of my review, but please let me know if i missed something i need to do to keep github happy

labkey-jeckels commented:
@bbimber nope, that edit looks fine to me and is part of the PR now.

labkey-jeckels committed:
Fix ViralLoadAssayTest - restore WNPRC test code, updated to new DOM in 20.x line, and avoiding WNPRC_EHR query metadata if the module is deployed but not enabled in the container

labkey-jeckels committed:
Merge remote-tracking branch 'origin/20.6_fb_merge_from_wnprc18.3' into 20.6_fb_merge_from_wnprc18.3
